### ISSUE TITLE
Gradient Descent algorithm

### ISSUE THREAD

**Issue by valleyxht** [valleyxht/blog-of-valley#3]
aim: minimise a cost/loss function 
有关最优化的问题，就是在给定条件下最小化一个方程，因为这个方程的要求是最小化，所以叫loss function ，如果是最大化就叫目标方程了。

最直观的理解：一开始的候我们随机站在一个点，把他看成一座山，每一步，我们都以下降最多的路线来下山，那么，在这个过程中我们到达山底（最优点）是最快的，而上面的a，它决定了我们“向下山走”时每一步的大小，过小的话收敛太慢，过大的话可能错过最小值。

梯度的数学表达：

![equation](https://latex.codecogs.com/svg.image?%5Cnabla%20f(p)=%5Cleft%5B%5Cbegin%7Barray%7D%7Bc%7D%5Cfrac%7B%5Cpartial%20f%7D%7B%5Cpartial%20x_1%7D(p)%5C%5C%5Cvdots%5C%5C%5Cfrac%7B%5Cpartial%20f%7D%7B%5Cpartial%20x_n%7D(p)%5Cend%7Barray%7D%5Cright%5D)
